### PR TITLE
Release 2.0.0-beta

### PR DESCRIPTION
**Pull request by IAlibay**
Merge develop into master & sets for release 2.0.0-beta.1

Changes made in this Pull Request:
 - This will make `master` 1 commit ahead of `develop`

Note: I'm using a semver beta release tag here, I'm not sure how well it'll play with pypi, but we'll see what happens with testpypi.

I did a testpypi release of 2.0.0a1 and everything works fine (except from the semver tests failing).

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
@richardjgowers is this pretty much the type of release pattern you were thinking of? (wasn't sure how annoying it would be to have `master` consistently ahead of `develop` by a few commits).

**Review by richardjgowers — APPROVED** [review]
If the missing date is deliberate we can merge this and do the beta release

File: package/CHANGELOG [hunk 1]
```diff
@@ -13,6 +13,330 @@ The rules for this file:
   * release numbers follow "Semantic Versioning" http://semver.org
 
 ------------------------------------------------------------------------------
+??/??/?? tylerjereddy, richardjgowers, IAlibay, hmacdope, orbeckst, cbouy,
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply package/CHANGELOG:16]: Yeah this is a problem since it's not a full release technically we shouldn't be tying off 2.0.0 in the changelog :/

**Comment by IAlibay** [comment]
I'm closing for now - see discord for discussions.

**Pull request closed without merging**